crates/bevy_geometry/src/lib.rs

```diff
@@ -0,0 +1,383 @@
+use bevy_math::*;
+use bevy_reflect::Reflect;
+use std::error::Error;
+use std::fmt;
+
+pub trait Primitive3d {
+    /// Returns true if this primitive is entirely on the outside (in the normal direction) of the
+    /// supplied plane.
+    fn outside_plane(&self, plane: Plane) -> bool;
+}
+
+#[derive(Debug, Clone)]
+pub enum PrimitiveError {
+    MinGreaterThanMax,
+    NonPositiveExtents,
+}
+impl Error for PrimitiveError {}
+impl fmt::Display for PrimitiveError {
```

Comment:
```suggestion
}

impl Error for PrimitiveError {}

impl fmt::Display for PrimitiveError {
```

just a small nit-pick, add some newlines here :)

Reply:
(Also this is here in a few other places, so adding newlines between impls would be nice)